docs/vector-library.md

```diff
@@ -0,0 +1,123 @@
+# Vector library
+
+## Summary
+
+Implement a standard library that provides functionality for the vector type.
+
+## Motivation
+
+Currently, vectors are a primitive type implemented internally. All of the heavy work to implement vectors is done, but there is no runtime-agnostic way to use vectors in Luau code. Individual Luau runtimes, such as [Lune](https://github.com/lune-org/lune) or Roblox, must provide their own distinct libraries to work with native vectors which may be inconsistent. In cross-runtime code, this results in performance drawbacks & difficulty utilizing native vectors.
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions & constants as a starting point, but as with all builtin libraries, future RFCs can propose additional functions.
+
+---
+
+`vector(x: number, y: number, z: number)`
+
+Creates a vector with 3 components: x, y, z. If the feature flag for wide vectors is enabled, a fourth argument `w: number?` will be introduced.
+
+Due to the common usage of vectors, vector creation should be ergonomic. Therefore, it is probably worth breaking the `create()` naming standard.
+
+`vector.magnitude(vecA: vector): number`
+
+Calculates the magnitude of a given vector.
+
+`vector.normalized(vec: vector): vector`
+
+Returns the normalized version (aka unit vector) of a given vector. If a zero vector is passed, return zero. For further clarification, `vector.normalized(vector(0, 0, 0))` should return zero.
+
+`vector.cross(vecA: vector, vecB: vector): vector`
+
+Returns the cross product of two vectors. If 4-wide vectors are enabled, this function will ignore the fourth component, and return the 3-dimensional cross product.
+
+`vector.dot(vecA: vector, vecB: vector): number`
+
+Returns the dot product of two vectors.
```

Comment:
Can we note how it behaves for 4-wide vectors?

Reply:
I'm unsure of the behavior it should have - I would simply use only the first three components even when compiled in four component mode. The other options are to error or compute the dot product of vectors with four components, which a quick google returns little results for.

Reply:
I don't think having it only work on the first 3 components is correct.
If we were to follow the definition of the dot product between 2 vectors
$$a \cdot b = \sum_{i=1}^n a_i b_i$$
we can say that the dot product for four component vectors would be:
$$\vec{a} \cdot \vec{b} = a_1 b_1 + a_2 b_2 + a_3 b_3 + a_4 b_4 $$

Reply:
resolved